src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -453,8 +445,11 @@ async def get_repository_pex(
             "`[python].resolve_all_constraints` is enabled, so "
             "`[python].requirement_constraints` must also be set."
         )
-    elif request.resolve_and_lockfile:
-        resolve, lockfile = request.resolve_and_lockfile
+    elif python_setup.enable_resolves:
+        # TODO: compute the resolve based on request.addresses, and validate that the
+        #  transitive closure is compatible. See coursier_fetch.py for inspiration.
```

Comment:
Should this point to a particular followup issue?

Reply:
Normally, yeah, but I plan to land this change tomorrow :) I wanted to simplify things first, including an upcoming PR to kill `generate-user-lockfiles`